src/Http/Routing.Abstractions/src/RoutingHttpContextExtensions.cs

```diff
@@ -46,8 +47,7 @@ public static object GetRouteValue(this HttpContext httpContext, string key)
                 throw new ArgumentNullException(nameof(key));
             }
 
-            var routingFeature = httpContext.Features[typeof(IRoutingFeature)] as IRoutingFeature;
-            return routingFeature?.RouteData.Values[key];
+            return httpContext.Features.Get<IRouteValuesFeature>()?.RouteValues[key];
```

Comment:
This doesn't really work because you expect `RouteData.Values` to be the source of truth. This implementation allows them to be different collections.  

`RouteData` is actually pretty complex - it is used to push and pop state (by MVC) which won't trigger updates to `httpContext.Request.RouteValues`. Trying to do a faithful job here would require some changes inside `RouteData`. 

I'd suggest just leaving this API alone and even marking it obsolete. If you're using old stuff, do `RouteData.Values`. If you're doing new stuff use `httpContext.Request.RouteValues`. 

I think if we're going to do more here than just leave this alone it will need some more design work.

Reply:
Now that I think about this more I'm objecting to it less. This only really causes problems when you use both sets of APIs (new and old) and use the old ones to set state, and the new ones to read it.

Reply:
Right, I think this API should just “work”. I’d the 3 collections get out of sync it’s be there was mutation involved and maybe we should update those places to also implement this feature to respect the mutation. I know this is all outside of the scope of this PR but I guess that’s why get for touching IRoutingFeature